functions/webhooks/taskrouterCallback.protected.ts

```diff
@@ -79,15 +67,41 @@ const isCleanupCustomChannel = (eventType: EventType, taskAttributes: { channelT
   return channelToFlex.isAseloCustomChannel(taskAttributes.channelType);
 };
 
+/**
+ * Fetch all taskrouter listeners from the listeners folder
+ */
+const getListeners = () => {
+  const functionsMap = Runtime.getFunctions();
+  const keys = Object.keys(functionsMap).filter((name) => name.includes(LISTENERS_FOLDER));
+  const paths = keys.map((key) => functionsMap[key].path);
+  return paths.map((path) => require(path) as TaskrouterListener);
+};
+
+const runTaskrouterListeners = async (
+  context: Context<EnvVars>,
+  event: EventFields,
+  callback: ServerlessCallback,
+) => {
+  const listeners = getListeners();
+
+  await Promise.all(
+    listeners
+      .filter((listener) => listener.shouldHandle(event))
+      .map((listener) => listener.handleEvent(context, event, callback)),
+  );
+};
+
 export const handler = async (
   context: Context<EnvVars>,
-  event: Body,
+  event: EventFields,
   callback: ServerlessCallback,
 ) => {
   const response = responseWithCors();
   const resolve = bindResolve(callback)(response);
 
   try {
+    await runTaskrouterListeners(context, event, callback);
+
     const { EventType: eventType } = event;
```

Comment:
Perhaps we can refactor this existing logic into TaskRouterListeners as well? Perhaps for another PR...

Reply:
Yeah, the plan to do that is described at the PR description